pkg/controllers/core/deployer_controller.go

```diff
@@ -613,8 +636,37 @@ func (r *DeployerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&corev1.Service{}).
 		Owns(&networkingv1beta1.Ingress{}).
 		// watch for build mutations to update dependent deployers
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, r.enqueueTrackedConfigMapResource()).
 		Watches(&source.Kind{Type: &buildv1alpha1.Application{}}, enqueueTrackedResources(&buildv1alpha1.Application{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Container{}}, enqueueTrackedResources(&buildv1alpha1.Container{})).
 		Watches(&source.Kind{Type: &buildv1alpha1.Function{}}, enqueueTrackedResources(&buildv1alpha1.Function{})).
 		Complete(r)
 }
+
+func (r *DeployerReconciler) constructHostForDeployer(deployer *corev1alpha1.Deployer, cm corev1.ConfigMap) string {
+	domain := defaultDomain
+	if d, ok := cm.Data[defaultDomainKey]; ok {
```

Comment:
handle case when valve is empty

```suggestion
	if d := cm.Data[defaultDomainKey]; d != "" {
```